backend/optimize.ml

```diff
@@ -0,0 +1,210 @@
+[@@@ocaml.warning "+a-29-40-41-42-4"]
+
+module DLL = Flambda_backend_utils.Doubly_linked_list
+
+module type IntCell = sig
+  include Csv_data.Cell
+
+  val update : t -> t
+end
+
+module IntCell : IntCell = struct
+  type t = int
+
+  let to_string = Int.to_string
+
+  let empty () = 0
+
+  let update t = t + 1
+end
+
+module IntCsv = Csv_data.Make (IntCell)
+
+let csv_singleton = ref Option.None
+
+let random_hex_string () =
+  let random_value = Random.int (1 lsl 24) in
+  Printf.sprintf "%06x" random_value
+
+let set_csv () =
+  if Option.is_none !csv_singleton
+  then (
+    let new_csv = IntCsv.create ["remove_useless_mov"; "fold_intop_imm"] in
+    if !Flambda_backend_flags.cfg_peephole_optimize_track
+    then Stdlib.at_exit (fun () -> 
+      Random.self_init (); 
+      (* Generate a random 6-hexdigit string for the name of the csv file *)
+      IntCsv.print new_csv (random_hex_string () ^ ".csv"));
```

Comment:
For testing it would be useful to
control where the file will be
created (i.e. parent folder). That
can be a parameter to the
command-line argument.